Test

AshwiniMunje opened the pull request.

Commit:
ToDo spring boot application: Sample app

Commit:
ToDo application: Add CRUD operations

This change includes:
1. Structural change - Addition of service and DAO layer
2. Addition of CRUD operations

uttpal commented on ToDo/src/main/java/com/todo/demo/Note/Note.java:
```diff
@@ -0,0 +1,58 @@
+
+//Mongo database annotation -- Model Class
+@Document("Note")
+public class Note {
```
use lombok, check lombok on internet and check @Data, @AllargsConstructor

AshwiniMunje replied:
Checked and updated.

uttpal commented on ToDo/src/main/java/com/todo/demo/Note/Note.java:
```diff
@@ -0,0 +1,58 @@
+        this.priority = priority;
+    }
+
+    public Integer getId() {
```
remove created setters and getters

AshwiniMunje replied:
Done.

uttpal commented on ToDo/src/main/java/com/todo/demo/Note/Priority.java:
```diff
@@ -0,0 +1,5 @@
+package com.todo.demo.Note;
+
+public enum Priority {
+    low, medium, high;
```
make enums in caps eg, LOW,MEDIUM,HIGH

AshwiniMunje replied:
Done

uttpal commented on ToDo/src/main/java/com/todo/demo/Resource/NoteResource.java:
```diff
@@ -0,0 +1,53 @@
+
+    //private final Logger logger = (Logger) LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
```
do not use field constructor use constructor injection (check on internet)

uttpal commented on ToDo/src/main/java/com/todo/demo/Service/NoteService.java:
```diff
@@ -0,0 +1,24 @@
+    //Create
+    public void createNote(Note note);
+
+    //Read
```
comments are not needed where use is implicit

AshwiniMunje replied:
Okay.

uttpal commented on ToDo/src/main/java/com/todo/demo/Service/NoteServiceImplementation.java:
```diff
@@ -0,0 +1,43 @@
+public class NoteServiceImplementation implements NoteService {
+
+    @Autowired
+    NoteRepository noteRepository;
```
create Dao and call repository from there

Commit:
ToDo: Address review comments

Commit:
ToDo: Add Dao layer

Add dao layer as suggested in review comments.

Commit:
ToDo: Add thrift file and client-server code

Commit:
ToDo: Update readNote and priority settings

uttpal commented on ToDo/src/main/java/com/todo/demo/Note/Note.java:
```diff
@@ -19,11 +20,50 @@
+    }
+
+    public Note toNote(NoteThrift noteThrift) {
+        Note note = new Note();
```
dont use empty constructor, check @AllArgsConstructor and @Builder in lombok project

AshwiniMunje replied:
Removed

uttpal commented on ToDo/src/main/java/com/todo/demo/Dao/NoteDao.java:
```diff
@@ -0,0 +1,20 @@
+
+public interface NoteDao {
+
+    int saveNote(Note note);
```
please return saved notes

AshwiniMunje replied:
Done.

uttpal commented on ToDo/src/main/java/com/todo/demo/Dao/NoteDao.java:
```diff
@@ -0,0 +1,20 @@
+public interface NoteDao {
+
+    int saveNote(Note note);
+    List<Note> read();
```
call this list notes, and add pagination and sorting

AshwiniMunje replied:
Renamed. I will look into the latter ones

uttpal commented on ToDo/src/main/java/com/todo/demo/Dao/NoteDao.java:
```diff
@@ -0,0 +1,20 @@
+    int saveNote(Note note);
+    List<Note> read();
+    void update(Note note);
+    void deleteNote(int id);
```
return deleted note

AshwiniMunje replied:
Done

uttpal commented on ToDo/src/main/java/com/todo/demo/Dao/NoteDao.java:
```diff
@@ -0,0 +1,20 @@
+
+    int saveNote(Note note);
+    List<Note> read();
+    void update(Note note);
```
return updated note

AshwiniMunje replied:
Done

uttpal commented on ToDo/src/main/Thrift/TodoThrift.thrift:
```diff
@@ -0,0 +1,29 @@
+     4:PriorityThrift priority;
+}
+
+service NoteServiceThrift {
```
return note object on each operation

AshwiniMunje replied:
Done

uttpal commented on ToDo/src/main/java/com/todo/demo/Note/Note.java:
```diff
@@ -19,11 +20,50 @@
+    public Note() {
+    }
+
+    public Note toNote(NoteThrift noteThrift) {
```
name it from fromTNote() and make it static function

AshwiniMunje replied:
Cool done

uttpal commented on ToDo/src/main/java/com/todo/demo/Service/NoteServiceHandler.java:
```diff
@@ -0,0 +1,59 @@
+    @Override
+    public int createNote(NoteThrift note) throws TException {
+        Note n = new Note();
+        n.toNote(note);
```
use static function then you don't have to create instance to call function

AshwiniMunje replied:
Done

uttpal commented on ToDo/src/main/java/com/todo/demo/Service/NoteServiceHandler.java:
```diff
@@ -0,0 +1,59 @@
+    public List<NoteThrift> readNote() throws TException {
+        List<NoteThrift> noteThriftList = new ArrayList<>();
+
+        for (Note n:noteService.readNote()) {
```
instead could be done as `return noteService.read().stream().map(Note::toThriftNote).collect(Collectors.toList())`

AshwiniMunje replied:
Any specific advantage of using this?

Commit:
ToDo: Address review comments

Following are the changes here:
1. Add return type for all functions
2. Modify constructors and add static functions

AshwiniMunje committed:
ToDo: Add pagination and sortinG

AshwiniMunje committed:
ToDo: Update pagination

AshwiniMunje committed:
ToDo: Add pagination in controller
